invenio/ext/babel/__init__.py

```diff
@@ -102,6 +102,21 @@ def set_translations():
     g._ = gettext
 
 
+def get_lazystring_encoder(app):
+    """Add a custom JSONEncoder that handles lazy strings from Babel."""
+    from speaklater import _LazyString
+    from six import text_type
+
+    class JSONEncoder(app.json_encoder):
+        def default(self, o):
+            if isinstance(o, _LazyString):
+                return text_type(o)
+
+            return app.json_encoder.default(self, o)
```

Comment:
either `return super(JSONEncoder, self).default(o)` or we need to keep a reference to the original `app.json_encoder`.